core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -49,6 +49,10 @@ object ExternalStoreUtils {
   final val DEFAULT_SAMPLE_TABLE_BUCKETS = "53"
   final val DEFAULT_TABLE_BUCKETS_LOCAL_MODE = "11"
   final val DEFAULT_SAMPLE_TABLE_BUCKETS_LOCAL_MODE = "7"
+  final val INDEX_TYPE = "INDEX_TYPE"
+  final val PREFIX_INDEX_TABLE = "SNCIDXTBL_"
+  final val INDEX_NAME = "INDEX_NAME"
+
```

Comment:
Lets use a common convention for prefixing so that it can be documented as out of bounds for users. Column tables uses ColumnFormatRelation.INTERNAL_SCHEMA_NAME. So index can use similar internal schema with some suffix to this (e.g. SNAPPYSYS_INTERNAL_INDEX). Then we can document that everything in SNAPPYSYS_INTERNAL* schema is out of bounds.